ext/src/http/client/curl/http_operation_curl.cc

```diff
@@ -1037,21 +1041,22 @@ CURLcode HttpOperation::Setup()
     return rc;
   }
 
-  rc = SetCurlPtrOption(CURLOPT_PROGRESSDATA, (void *)this);
+  rc = SetCurlPtrOption(CURLOPT_PROGRESSDATA, this);
   if (rc != CURLE_OK)
   {
     return rc;
   }
 #endif
 
 #if LIBCURL_VERSION_NUM >= 0x075000
-  rc = SetCurlPtrOption(CURLOPT_PREREQFUNCTION, (void *)&HttpOperation::PreRequestCallback);
+  rc = SetCurlPtrOption(CURLOPT_PREREQFUNCTION,
+                        reinterpret_cast<void *>(&HttpOperation::PreRequestCallback));
   if (rc != CURLE_OK)
   {
     return rc;
   }
 
-  rc = SetCurlPtrOption(CURLOPT_PREREQDATA, (void *)this);
+  rc = SetCurlPtrOption(CURLOPT_PREREQDATA, this);
```

Comment:
Should `static_cast<void *>` be added here?

Reply:
No, no need for it, SetCurlPtrOption() is a otel-cpp helper with proper types, not a curl api.

Reply:
But `SetCurlPtrOption` accepts `void *` instead of the typed pointer of `this`, so we assume it is safe to convert `this` to `void *` type?

Reply:
Any pointer `T *` can be converted to `void *`, no need for a cast.